web/app/js/components/Sidebar.jsx

```diff
@@ -148,54 +204,31 @@ class Sidebar extends React.Component {
               </PrefixedLink>
             </Menu.Item>
 
-            <Menu.Item className="sidebar-menu-item" key="/namespaces">
-              <PrefixedLink to="/namespaces">
-                <Icon type="dashboard" />
-                <span>Namespaces</span>
-              </PrefixedLink>
-            </Menu.Item>
-
-            <Menu.Item className="sidebar-menu-item" key="/tap">
-              <PrefixedLink to="/tap">
-                <Icon type="filter" />
-                <span>Tap</span>
-              </PrefixedLink>
-            </Menu.Item>
-
             {
-              _.map(_.take(this.state.namespaces, this.state.maxNsItemsToShow), ns => {
+              _.map(_.keys(sidebarComponents).sort(), resourceName => {
                 return (
-                  <Menu.Item className="sidebar-submenu-item" key={`/namespaces/${ns}`}>
-                    <PrefixedLink to={`/namespaces/${ns}`}>
-                      <Icon>{this.state.collapsed ? _.take(ns, 2) : <span>&nbsp;&nbsp;</span> }</Icon>
-                      <span>{ns} {this.state.collapsed ? "namespace" : ""}</span>
-                    </PrefixedLink>
-                  </Menu.Item>
+                  <SubMenu
```

Comment:
if there are no sub items in the menu, we should either omit or disable linking for the high level component (e.g. there aren't any deployments when you select namespace `kube-system` and so the nave is misleading; it takes a lot of time to click on each resource type and figure out that there aren't resources in it)